domain/application/state/application_test.go

```diff
@@ -237,6 +240,217 @@ func (s *applicationStateSuite) TestCreateApplicationWithCharmStoragePath(c *gc.
 	s.assertApplication(c, "666", platform, channel, scale, true)
 }
 
+// TestCreateApplicationWithResources tests creation of an application with
+// specified resources.
+// It verifies that the charm_resource table is populated, alongside the
+// resource and application_resource table with datas from charm and arguments.
+func (s *applicationStateSuite) TestCreateApplicationWithResources(c *gc.C) {
```

Comment:
This test is fairly verbose, might be worth splitting out the SQL queries into their own functions to make it more readable.

Reply:
That's a good remark. I don't think that creating more function in a UT always make the test more readable. 

It is always the question between cluttered context and loose context. In one case it is painful to understand the test because there are too much character with few signification. On the other case it is painful because usually, function name doesn't catch what they really do (and sometimes, check), and you have to scroll in the UT to understand what is going one.

I usually prefer having more information directly in the UT, following this rule of thumb:

* Only Relevant data (the data that will change the test behavior if changed. Usually data that vary between similar UT)
* if there is required data (not relevant, but required), hide them or mark them (`// the name doesn't matter`)
* No other data (if a field can be empty/nil without break the test, leave it empty/nil
* If it break, give context (use `gc.Commentf`) and try to have a link as close as possible on the faulty test
   * factorizing assert in method usually break this rule
   * an assert that check some "Arrange phase" (db populating, for instance), has not the same meaning than one which actually check something in the test (Assert phase). The former means the test is broken, the latter means that the functionality is broken. 
   
   I've tried to follow those rules, and I am not convinced that moving the sql request out will help, because... It is basically what we want to test (the state of the db after create application)